tests/src-android/cgeo/geocaching/models/GeocacheTest.java

```diff
@@ -70,12 +70,20 @@ public final void testUpdateWaypointFromNote() {
         assertWaypointsParsed("Test N51 13.888 E007 03.444", wpList);
     }
 
+    public final void testUpdateWaypointsFromNoteSingleLine() {
+        final List<Waypoint> wpList = new ArrayList<>();
+        wpList.add(createWaypointWithUserNote(new Geopoint("N51 13.888 E007 03.444"), "", "", "Test N51 13.233 E007 03.444 Test N51 09.123 E007 03.444", WaypointType.OWN));
+        wpList.add(createWaypointWithUserNote(new Geopoint("N51 13.233 E007 03.444"), "", "", "Test N51 09.123 E007 03.444", WaypointType.OWN));
+        wpList.add(createWaypointWithUserNote(new Geopoint("N51 09.123 E007 03.444"), "", "", "", WaypointType.OWN));
+        assertWaypointsParsed("Test N51 13.888 E007 03.444 Test N51 13.233 E007 03.444 Test N51 09.123 E007 03.444",  wpList);
```

Comment:
Also newlines beetween waypoints as is testUpdateWaypointsFromNote/testUpdateWaypointsFromNoteWithDuplicates?

Reply:
No, that was on purpose (therefor the name SingleLine) to have the waypoints in a single line, so it shows, the behaviour of extracting the user note -> within a single line, the user note for the first waypoints contains the following waypoints
-> I will add a test-description